### PR TITLE
[flink]add spatial type for mysql cdc action

### PR DESCRIPTION
**Pull request by zhangjun0x01**
*(Please specify the module before the PR name: [core] ... or [flink] ...)*

### Purpose

**issue https://github.com/apache/incubator-paimon/issues/966 **

add the follow data type:

GEOMETRY
POINT
LINESTRING
POLYGON
MULTIPOINT
MULTILINESTRING
MULTIPOLYGON
GEOMETRYCOLLECTION

### Tests

*org.apache.paimon.flink.action.cdc.mysql.MySqlSyncTableActionITCase*

### API and Format 

*(Does this change affect API or storage format)*

### Documentation

*(Does this change introduce a new feature)*

### CODE REVIEW

File: paimon-flink/paimon-flink-common/pom.xml [hunk 1]
```diff
@@ -81,6 +82,18 @@ under the License.
...
+        <dependency>
+            <groupId>com.esri.geometry</groupId>
+            <artifactId>esri-geometry-api</artifactId>
+            <version>${geometry.version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>com.fasterxml.jackson.core</groupId>
+                    <artifactId>jackson-core</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/pom.xml:94]: provided?

File: paimon-flink/paimon-flink-common/pom.xml [hunk 2]
```diff
@@ -81,6 +82,18 @@ under the License.
             <version>${frocksdbjni.version}</version>
         </dependency>
 
+        <dependency>
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/pom.xml:85]: Put this together with cdc dependency, and add comment for what.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java [hunk 3]
```diff
@@ -200,8 +203,10 @@ public List<CdcRecord> getRecords() {
     }
 
     private Map<String, String> extractRow(JsonNode recordRow) {
-        Map<String, String> recordMap =
-                objectMapper.convertValue(recordRow, new TypeReference<Map<String, String>>() {});
+
+        // the geometry,point type can not be converted to string, so we convert it to Object first.
+        Map<String, Object> recordMap =
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java:208]: Can we just rename this to `jsonMap`?
And create a new map `Map<String, String> resultMap`.

Comment (zhangjun0x01) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java]: I update all comments .

**Comment by zhangjun0x01** [comment]
image

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java [hunk 4]
```diff
@@ -200,18 +203,27 @@ public List<CdcRecord> getRecords() {
...
+        // the geometry,point type can not be converted to string, so we convert it to Object first.
+        Map<String, Object> jsonMap =
+                objectMapper.convertValue(recordRow, new TypeReference<Map<String, Object>>() {});
+
+        if (jsonMap == null) {
             return new HashMap<>();
         }
 
+        Map<String, String> resultMap = new HashMap<>();
+        for (Map.Entry<String, Object> field : jsonMap.entrySet()) {
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java:216]: Do we need this?

Comment (zhangjun0x01) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java:216]: This is for the convenience of handling the subsequent logic. The `oldValue` required  in the for loop is a string type, and the return value is `Map<String, String>`. Therefore, we need to convert `Map<String, Object>` to `Map<String, String>`

Comment (JingsongLi) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java:216]: I know, but subsequent for loop will assign each key? Why assign again?

**Comment by zhangjun0x01** [comment]
image

**Comment by zhangjun0x01** [comment]
image


there is an exception in core module ,I fix it  in other pr https://github.com/apache/incubator-paimon/pull/1060
```

Error:  Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 6.145 s <<< FAILURE! - in org.apache.paimon.operation.PartitionExpireTest
Error:  testFilterCommittedAfterExpiring  Time elapsed: 5.714 s  <<< ERROR!
java.time.format.DateTimeParseException: Text '20230499' could not be parsed: Invalid value for DayOfMonth (valid values 1 - 28/31): 99
	at java.time.format.DateTimeFormatter.createError(DateTimeFormatter.java:1920)
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1855)
	at java.time.LocalDate.parse(LocalDate.java:400)
	at org.apache.paimon.partition.PartitionTimeExtractor.toLocalDateTime(PartitionTimeExtractor.java:112)
	at org.apache.paimon.partition.PartitionTimeExtractor.extract(PartitionTimeExtractor.java:97)
	at org.apache.paimon.operation.PartitionExpire.doExpire(PartitionExpire.java:95)
	at org.apache.paimon.operation.PartitionExpire.expire(PartitionExpire.java:85)
	at org.apache.paimon.operation.PartitionExpire.expire(PartitionExpire.java:74)
	at org.apache.paimon.table.sink.TableCommitImpl.expire(TableCommitImpl.java:145)
	at org.apache.paimon.table.sink.TableCommitImpl.commit(TableCommitImpl.java:105)
	at org.apache.paimon.operation.PartitionExpireTest.testFilterCommittedAfterExpiring(PartitionExpireTest.java:189)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.jav


```

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**